### PR TITLE
Adding draft SLA to issue-management.md file

### PR DESCRIPTION
**Pull request by thomashchan1**
Added draft SLA for issues and PRs in CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,4 +18,41 @@ OK on GitHub.
...
+the box settings for this repository will be consistent.
+
+## SLA
+
+For Issues
+
+* For original poster (OP), after opening an urgent issue, please allow at least 
+(proposed) 1 business day before following up with the triager or assignee to 
+give time to work the issue and respond. Feel free to provide additional info/data 
+or ask further questions at anytime. Use URGENT label to distinguish from other
```
Comment (songy23) [inline CONTRIBUTING.md]: We use P0, P1, P2 labels in OpenCensus-Specs to indicate the priority and urgency: https://github.com/census-instrumentation/opencensus-specs/blob/master/triage-process.md#issue-handling-process. IMO that's more organized than having one "urgent" label.

/cc @dinooliva @mayurkale22 @rghetia

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -18,4 +18,41 @@ OK on GitHub.
...
+issues and provide reason behind the urgency.
+
+* For OP, after opening a non-urgent issue, please allow at least (proposed) 3 
+business days before follow up with the triager or assignee to give time to work 
+the issue and respond. Feel free to provide additional info/data or ask further 
+questions at anytime.
+
+* For assignee, if the associated PR is merged, please proceed to close the issue, 
+if not done already. For issues without PR and after providing 
+answers/resolution/workarounds, please allow at least (proposed) 2 business days
```
Comment (bogdandrutu) [inline CONTRIBUTING.md]: I think this will lead to tons of opened PRs. I would suggest that we are allow to close the PRs and people can reopen the PR if needed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (thomashchan1) [reply CONTRIBUTING.md]: Thanks, @songy23 . The original intention was to keep this SLA as simple as possible. If team feel we really need to break issues down into P0, P1 and P2, and spell out the criteria and SLA for each, we certainly can. Thoughts?

File: CONTRIBUTING.md (hunk 2 above)
Comment (thomashchan1) [reply CONTRIBUTING.md]: Believe this section is related to issues without PRs. Those with PRs would be closed as the PRs are merged. Don't think this would leave a lot of issues open. The 2 business day wait is aligned with RFC final closure as well. HTH.

File: CONTRIBUTING.md (hunk 1 above)
Comment (songy23) [reply CONTRIBUTING.md]: I would prefer to have finer-grained on priority levels. For different priorities, SLAs should be different too. Would like to hear opinions on this.

Comment (thomashchan1) [reply CONTRIBUTING.md]: Anyone else has input on this? My take is to start small and simple with urgent and non-urgent classification. If we see cases that require more granular priority levels, we can certainly do that in the future. Want to move forward, at least make incremental effort to state high level SLAs to avoid us stepping on each others toes.

File: CONTRIBUTING.md (hunk 2 above)
Comment (thomashchan1) [reply CONTRIBUTING.md]: @bogdandrutu, do you have other concerns on this? Following the draft SLA to see if we can move this forward. Thanks in advance.

File: CONTRIBUTING.md (hunk 1 above)
Comment (iredelmeier) [reply CONTRIBUTING.md]: I strongly believe that rather than labels, we should handle this directly through prioritization. In particular, GitHub [project boards](https://help.github.com/en/articles/about-project-boards) let us turn issues into a backlog. If something higher priority comes up, we can simply drag it above lower priority issues.

"Urgent" is probably very useful for security issues. On the other hand, security triage may warrant a separate discussion altogether, since it is one thing that expressly needs to occur behind closed doors at least some of the time so that attackers cannot take advantage of known vulnerabilities before they're patched. Traditional P0, P1, P2, etc labeling, on the other hand, is hard to map to a cross-company OSS project, since those labels tend to be based on customer needs and we all inherently have different customers (and therefore care about different customer needs).

File: CONTRIBUTING.md (hunk 2 above)
Comment (reyang) [reply CONTRIBUTING.md]: I suggest that we close the issue when we got the PR/fix merged and package published. This is a consistent signal for OP and people who are following the issue. OP can always reopen the issue if there is a need.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -18,4 +18,41 @@ OK on GitHub.
...
+give time to work the issue and respond. Feel free to provide additional info/data 
+or ask further questions at anytime. Use URGENT label to distinguish from other 
+issues and provide reason behind the urgency.
+
+* For OP, after opening a non-urgent issue, please allow at least (proposed) 3 
+business days before follow up with the triager or assignee to give time to work 
+the issue and respond. Feel free to provide additional info/data or ask further 
+questions at anytime.
+
+* For assignee, if the associated PR is merged, please proceed to close the issue,
```
Comment (reyang) [inline CONTRIBUTING.md]: Do we always fix in the latest version, or we might need to do a hotfix on an old version?
For example, if we're on version 1.5, while people reported a severe memory leak on version 1.2, do we fix 1.2+ or we fix it in 1.5.1?
What about major versions? If we're on 2.0, do we still provide SLA for 1.0?

Comment (rghetia) [reply CONTRIBUTING.md]: In addition to `priority` if there was `severity` (Critical, Major, Minor, Trivial) of the issue then Critical fixes should go in older major releases. For example any security bug fix would fall in Critical category.

Additionally we would need End-of-Life guidelines for major versions.

File: CONTRIBUTING.md (hunk 2 above)
Comment (thomashchan1) [reply CONTRIBUTING.md]: Believe that's the plan. Let me know if this is still unclear.

File: CONTRIBUTING.md (hunk 3 above)
Comment (thomashchan1) [reply CONTRIBUTING.md]: @iredelmeier will move this from contributing.md to a new SLA.md. As discussed earlier, we would try this in the spec repo for couple weeks. If it works, will move to the community repo to be applicable to all SIGs, as appropriate.

@reyang I didn't specify this in the initial SLA, as that was just a start for handling issues and PRs. My initial take is that fixing in the latest version is the default and asking users to upgrade to the latest would be the recommended path. Propagating the fix to prior versions would be optional and be considered on a case by case basis based on criticality of the issue, per @rghetia. Tried to keep things simple by not introducing new labels/classifications unless it's necessary. Will think this one through.

@rghetia again I did not go that far in defining EOL guidelines. This is probably best to be worked when we finalize our compatibility/backward compatibility strategy. Thoughts?

Comment (rghetia) [reply CONTRIBUTING.md]: > @rghetia again I did not go that far in defining EOL guidelines. This is probably best to be worked when we finalize our compatibility/backward compatibility strategy. Thoughts?

EOL guidelines can be separate and we don't need to define it now. However, we should do that before making GA release.

Comment (thomashchan1) [reply CONTRIBUTING.md]: Sounds good. Thanks, @rghetia

**Comment by thomashchan1** [comment]
@AloisReitbauer, @bogdandrutu, @c24t, @reyang, @SergeyKanzhelev, @songy23, @rghetia and/or @yurishkuro, please review the update SLA changes, namely:
1. moved changes to a new issue-management.md file instead of in contributing.md
2. simplified the triage documentation from OC: https://github.com/census-instrumentation/opencensus-specs/blob/master/triage-process.md and also from backlog management PR: https://github.com/open-telemetry/community/pull/52, keeping priorities and labels to the minimum.

File: issue-management.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  - Person whose Approval is needed to merge the PR.
+
+
+## Opening an Issue
+
+- An issue is filed by OP.
+- A Maintainer responds and asks clarifying questions within 1-2 business days.
+- The Maintainer processes the issue and lables it as:
+  - `bug`
+  - `enhancement`
```
Comment (songy23) [inline issue-management.md]: and `feature-request`?

File: issue-management.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    are responsible for making sure issue status labels are up to date.
+- Reviewer:
+  - Person whose Approval is needed to merge the PR.
+
+
+## Opening an Issue
+
+- An issue is filed by OP.
+- A Maintainer responds and asks clarifying questions within 1-2 business days.
+- The Maintainer processes the issue and lables it as:
```
Comment (reyang) [inline issue-management.md]: lables -> labels?

File: issue-management.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  - `bug`
```
Comment (reyang) [inline issue-management.md]: Ideally we don't need to label it manually, the issue template can help a lot here.

File: issue-management.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+
+## Closing an Issue
+
+- Review criteria:
+  - For interface and design changes: 2 approvals - which must be from reviewers 
+    who work at different companies than the Collaborator.
+  - For smaller or internal changes: 1 approval from a different company.
+- For `URGENT` issues:
+  - Collaborator: please provide initial assessment of the issues to OP ASAP or
```
Comment (reyang) [inline issue-management.md]: Do we intend to fix the issue in the next release, or we want to do hotfix? Any guidance on branching and release process? (might be out of the scope for this PR though)

File: issue-management.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+questions.
+
+
+## When Issues Get Stuck
+
+Some issues are not directly related to a particular code change. If an 
+issue is worth considering in the issue backlog, but not scoped clearly 
+enough for work to begin, then please label it `need-discussion`.
+
+- When possible, move the disucssion forward by using tests and code examples.
```
Comment (reyang) [inline issue-management.md]: disucssion  -> discussion

File: issue-management.md (hunk 4 above)
Comment (thomashchan1) [reply issue-management.md]: Thanks, @songy23 . Yes, let's start with this and we can iterate if we indeed need more granularity. Just didn't want to amplify the SLA with more priority levels unless we really need to. As for feature-request, the intention here again is keep things simple as enhancement should cover that. Thoughts?

File: issue-management.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+  issue.
+- When an agreement is made, clearly summarize the decision, and list any 
+  resulting action items which need to be addressed.
+  
+If an issue is stuck because someone is not responding, please add the `stale` 
+label. It is possible to automate this. E.g. https://github.com/apps/stale 
+The minimum time elapsed before the `stale` label is applied is proposed to be 
+one week.
+
+We highly encourage to use line breaks in markdown files at `80` characters
```
Comment (yurishkuro) [inline issue-management.md]: This seems completely unrelated to the  issue management

File: issue-management.md (hunk 6 above)
Comment (thomashchan1) [reply issue-management.md]: Thanks, @reyang Yes, we can certainly experiment with that.

File: issue-management.md (hunk 7 above)
Comment (thomashchan1) [reply issue-management.md]: Yeah, probably best to add that as part of a separate release process PR when we get to that point. Thanks for bringing that up though.

File: issue-management.md (hunk 8 above)
Comment (thomashchan1) [reply issue-management.md]: ah...will correct. Thanks.

File: issue-management.md (hunk 9 above)
Comment (thomashchan1) [reply issue-management.md]: Oh, the file format part? Will remove. Thought it was part of the template.

File: issue-management.md (hunk 5 above)
Comment (thomashchan1) [reply issue-management.md]: updated

File: issue-management.md (hunk 8 above)
Comment (thomashchan1) [reply issue-management.md]: Done

File: issue-management.md (hunk 9 above)
Comment (thomashchan1) [reply issue-management.md]: Removed

**Comment by rghetia** [comment]
LGTM

File: issue-management.md (hunk 4 above)
Comment (songy23) [reply issue-management.md]: I think this is good to start with. We can revisit it if priority and `feature-request` label is a more common practice.

**Comment by thomashchan1** [comment]
Thanks, @reyang and @songy23 

@yurishkuro would love to find ways to measure these via automation. If see if I can find anything that we can use. Anyone else has suggestions, please send them to our Gitter channel. In the interim, will need to depend on all of us to identify gaps, politely call them out so that we can address them and get better together.

File: issue-management.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+## Opening an Issue
+
+- An issue is filed by OP.
+- A Maintainer responds and asks clarifying questions within 1-2 business days.
+- The Maintainer processes the issue and labels it as:
+  - `bug`
+  - `enhancement`
+  - `need-discussion`
+  - `documentation` or
+  - `will-not-fix` (thereby closing the issue with explicit reasons)
```
Comment (c24t) [inline issue-management.md]: Not `wontfix`?

File: issue-management.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+- The Maintainer can also label the issue as
+  - `URGENT` (for critical issues)
+  - `help-wanted` for issues which require work and have no one assigned
+- Once a Collaborator is assigned, please remove `help-wanted` and add `wip` to 
+  the issue.
```
Comment (c24t) [inline issue-management.md]: Label style nits: `URGENT` gets capitalized but `wip` doesn't?

File: issue-management.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+## Closing an Issue
+
+- Review criteria:
+  - For interface and design changes: 2 approvals - which must be from reviewers 
+    who work at different companies than the Collaborator.
+  - For smaller or internal changes: 1 approval from a different company.
+- For `URGENT` issues:
+  - Collaborator: please provide initial assessment of the issues to OP ASAP or 
+    within 1 business day, whichever is earlier.
```
Comment (c24t) [inline issue-management.md]: "ASAP or within X" seems to have the same logic as "up to X% off or more" -- it could describe any number.

File: issue-management.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+  - Reviewer: please review and provide feedback ASAP or within 1 business day, 
+    whichever is earlier.
+  - Collaborator: please provide update and/or responses to review comments ASAP
```
Comment (c24t) [inline issue-management.md]:
```suggestion
  - Collaborator: please provide an update and/or responses to review comments ASAP
```

File: issue-management.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+  the issue.
+
+
+## Closing an Issue
+
+- Review criteria:
+  - For interface and design changes: 2 approvals - which must be from reviewers 
+    who work at different companies than the Collaborator.
+  - For smaller or internal changes: 1 approval from a different company.
+- For `URGENT` issues:
```
Comment (c24t) [inline issue-management.md]: Will this doc include info about classifying issues? Is `URGENT` reserved for e.g. security issues or is it anything a maintainer thinks is urgent?

File: issue-management.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+  - Collaborator: please provide initial assessment of the issues to OP ASAP or 
+    within 1 business day, whichever is earlier.
+  - Reviewer: please review and provide feedback ASAP or within 1 business day, 
+    whichever is earlier.
+  - Collaborator: please provide update and/or responses to review comments ASAP
+    or within 1 business day, whichever is sooner. Merge should happen as soon as
+    review criteria are met.
+- For non-`URGENT` issues
+  - Collaborator: please provide initial response or assessment of the issue to
```
Comment (c24t) [inline issue-management.md]:
```suggestion
  - Collaborator: please provide an initial response or assessment of the issue to 
```

File: issue-management.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+  - Reviewer: please review and provide feedback ASAP or within 1 business day, 
+    whichever is earlier.
+  - Collaborator: please provide update and/or responses to review comments ASAP
+    or within 1 business day, whichever is sooner. Merge should happen as soon as
+    review criteria are met.
+- For non-`URGENT` issues
+  - Collaborator: please provide initial response or assessment of the issue to 
+    OP within 3 business days.
+  - Reviewer: please review and provide feedback within 3 business days.
+  - Collaborator: please provide update and/or responses to review comments
```
Comment (c24t) [inline issue-management.md]:
```suggestion
  - Collaborator: please provide an update and/or responses to review comments
```

File: issue-management.md [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+  - Collaborator: please provide initial assessment of the issues to OP ASAP or 
+    within 1 business day, whichever is earlier.
+  - Reviewer: please review and provide feedback ASAP or within 1 business day, 
+    whichever is earlier.
+  - Collaborator: please provide update and/or responses to review comments ASAP
+    or within 1 business day, whichever is sooner. Merge should happen as soon as
+    review criteria are met.
+- For non-`URGENT` issues
+  - Collaborator: please provide initial response or assessment of the issue to 
+    OP within 3 business days.
```
Comment (c24t) [inline issue-management.md]: What happens after 3 business days with no response, but before 7?

File: issue-management.md [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+
+
+## Opening an Issue
+
+- An issue is filed by OP.
+- A Maintainer responds and asks clarifying questions within 1-2 business days.
+- The Maintainer processes the issue and labels it as:
+  - `bug`
+  - `enhancement`
+  - `need-discussion`
```
Comment (c24t) [inline issue-management.md]: I'd prefer the more-grammatical-sounding `needs-discussion`.

File: issue-management.md (hunk 10 above)
Comment (thomashchan1) [reply issue-management.md]: Thanks, @c24t this is just to keep it grammatically correct. No strong feeling either way though. To be consistent, if we change, it will be something like 'wont-fix'.

File: issue-management.md (hunk 11 above)
Comment (thomashchan1) [reply issue-management.md]: The intention is to keep everything lower case except for URGENT to raise attention. Again, either would work.

File: issue-management.md (hunk 12 above)
Comment (thomashchan1) [reply issue-management.md]: The intention here is to highlight the request for quick response while stating the upper limit. Let me know your thoughts.

File: issue-management.md (hunk 14 above)
Comment (thomashchan1) [reply issue-management.md]: At the moment, this would be anything the OP and/or maintainer believe is urgent. Nothing reserved for now.

File: issue-management.md (hunk 15 above)
Comment (thomashchan1) [reply issue-management.md]: Ah, need to make this consistent as well. Thanks.

File: issue-management.md (hunk 16 above)
Comment (thomashchan1) [reply issue-management.md]: Will make this consistent.

File: issue-management.md (hunk 17 above)
Comment (thomashchan1) [reply issue-management.md]: It would just sit there. If we can find automation tools that work, we can send reminders.

File: issue-management.md (hunk 12 above)
Comment (c24t) [reply issue-management.md]: How about something like this?

"Please provide an initial assessment of the issue within 1 business day. `URGENT` issues are time-sensitive. Collaborators should prioritize the assessment and responses to review comments above other work and respond as quickly as possible."

"As quickly as possible" still feels fishy to me since this is what SLAs are meant to solve.

File: issue-management.md (hunk 18 above)
Comment (thomashchan1) [reply issue-management.md]: Done

File: issue-management.md (hunk 12 above)
Comment (thomashchan1) [reply issue-management.md]: Understood @c24t . Given this is community based and not really the same as enforceable within one company, suspect this is as good as we can get in terms of prescribing the intent. One question is while this sounds better and more descriptive as well, do you (or anyone else) feel comfortable duplicating this, with some minor adjustment, 2 other times for URGENT issue SLA vs the existing shorter versions? For other minor comments above, I am planning to resolve them to move things along. Let me know otherwise or feel free to re-open if you have strong feelings on those. Thanks again for spending the time to provide input.

File: issue-management.md (hunk 16 above)
Comment (thomashchan1) [reply issue-management.md]: Done

File: issue-management.md (hunk 15 above)
Comment (thomashchan1) [reply issue-management.md]: Done

File: issue-management.md (hunk 11 above)
Comment (thomashchan1) [reply issue-management.md]: Resolving this for now. Let me know if you have strong feelings about this.

File: issue-management.md (hunk 10 above)
Comment (thomashchan1) [reply issue-management.md]: Keeping this as is for now. Let me know if you have strong feelings about this.

File: issue-management.md (hunk 12 above)
Comment (thomashchan1) [reply issue-management.md]: @c24t let me know your thoughts on this. Would like to get this merged soon. Thanks in advance.
